src/dune_pkg/single_run_file_cache.ml

```diff
@@ -1,14 +1,9 @@
 open! Import
-open! Stdune
 
-type entry =
-  { filename : Filename.t
-  ; complete : unit Fiber.Ivar.t
-  }
-
-type t =
+type 'error t =
   { temp_dir : Path.t lazy_t
-  ; key_to_filename : entry String.Table.t
+  ; key_to_filename : (string, (Path.t, 'error) result) Fiber_cache.t
+  ; num_entries : int ref
```

Comment:
what about just `mutable num_entries : int`? will reduce the extra laye of indirection.